subworkflows/local/profiling.nf

```diff
@@ -67,6 +67,14 @@ workflow PROFILING {
                     [meta + [id: "${meta.id}${meta.single_end ? '_se' : '_pe'}"], reads]
             }
             .combine(databases)
+            .filter { it ->
+                def platform = it[0]['instrument_platform']
+                def db_type = it[2]['db_type']
+                def is_long_read = platform == 'OXFORD_NANOPORE'
+                def is_long_db = db_type == 'long' || db_type == 'both'
+                def is_short_db = db_type == 'short' || db_type == 'both'
+                (is_long_read && is_long_db) || (!is_long_read && is_short_db)
```

Comment:
I don't really get this, what are you actually trying to filter for? Is this if you've bound a short-read dataset with long-read database, for example?

Reply:
Is it not possible to prepare all the info prior and combine just on that directly (rather than having the overhead of making all possible combinations and then just filtering to a few?)

Reply:
I have now combined the reads and db channels by `db_type` or `read_type` instead of filtering them.